packages/ui/src/ui/pages/tablet-errors/TabletErrorsToolbar.tsx

```diff
@@ -0,0 +1,130 @@
+    'Write',
+    'Trim',
+].map((value) => ({value, text: value}));
+
+export function TabletErrorsToolbar({bundle, className}: {bundle: string; className: string}) {
+    const dispatch = useDispatch();
+
+    const methodsFilter = useSelector(getTabletErrorsByBundleMethodsFilter);
+    const timeRangeFilter = useSelector(getTabletErrorsByBundleTimeRangeFilter);
+    const pageFilter = useSelector(getTabletErrorsByBundlePageFilter);
+    const {from, to} = useTabletErrorsLoad(bundle, {methodsFilter, timeRangeFilter, pageFilter});
+
+    return (
+        <div className={block(null, className)}>
+            <YTTimeline
+                from={from}
+                to={to}
+                shortcut={timeRangeFilter.shortcutValue}
+                onUpdate={(data) => {
+                    console.log({data});
+                    dispatch(tabletErrorsByBundleActions.updateFilter({timeRangeFilter: data}));
+                }}
+                hasRuler={true}
+            />
+            <Toolbar
+                itemsToWrap={[
+                    {
+                        node: (
+                            <SimplePagination
+                                value={pageFilter}
+                                min={0}
+                                max={100}
+                                onChange={(v) => {
+                                    dispatch(
+                                        tabletErrorsByBundleActions.updateFilter({pageFilter: v}),
+                                    );
+                                }}
+                            />
+                        ),
+                    },
+                    {
+                        node: (
+                            <Select
+                                multiple
+                                label="Methods:"
+                                value={methodsFilter}
+                                items={ALL_METHODS}
+                                onUpdate={(v) =>
+                                    dispatch(
+                                        tabletErrorsByBundleActions.updateFilter({
+                                            methodsFilter: v,
+                                        }),
+                                    )
+                                }
+                                placeholder="Select..."
+                                hasClear
+                                maxVisibleValuesTextLength={80}
+                            />
+                        ),
+                    },
+                ]}
+            />
+        </div>
+    );
+}
+
+function useTabletErrorsLoad(
+    bundle: string,
+    {
+        timeRangeFilter,
+        methodsFilter,
+        pageFilter,
+    }: Pick<TabletErrorsByBundleState, 'timeRangeFilter' | 'methodsFilter' | 'pageFilter'>,
+) {
+    const dispatch = useDispatch();
+
+    React.useEffect(() => {
+        const {from, to} = calcFromTo(timeRangeFilter);
+        dispatch(
```

Comment:
`loadTabletErrorsByBundle` has no error handling. Can I add them here?

```
function useTabletErrorsLoad(
    bundle: string,
    {
        timeRangeFilter,
        methodsFilter,
        pageFilter,
    }: Pick<TabletErrorsByBundleState, 'timeRangeFilter' | 'methodsFilter' | 'pageFilter'>,
) {
    const dispatch = useDispatch();
    const [error, setError] = React.useState<Error | null>(null);

    React.useEffect(() => {
        try {
            const {from, to} = calcFromTo(timeRangeFilter);
            dispatch(
                loadTabletErrorsByBundle(pageFilter, {
                    tablet_cell_bundle: bundle,
                    start_timestamp: Math.floor(from / 1000),
                    end_timestamp: Math.ceil(to / 1000),
                    methods: methodsFilter.length ? methodsFilter : undefined,
                }),
            );
        } catch (err) {
            setError(err);
        }
    }, [bundle, dispatch, timeRangeFilter, methodsFilter, pageFilter]);

    const {from, to} = React.useMemo(() => calcFromTo(timeRangeFilter), [timeRangeFilter]);

    return {from, to, error};
}
```

Reply:
It is better to add error handling to the action